src/arch/traits.rs

```diff
@@ -22,6 +39,15 @@ pub trait Registers: Default {
     fn gdb_deserialize(&mut self, bytes: &[u8]) -> Result<(), ()>;
 }
 
+/// Data that can be serialized according to GDB protocol.
+pub trait GdbSerializable {
```

Comment:
To keep things DRY, you could just make `Registers` a supertrait of `GdbSerializable` (i.e: `trait Registers: Default + GdbSerializable`). After all, why have two sets of methods that do the exact same thing :smile:

Reply:
I dropped the `GdbSerializable` trait in the new iteration.